### PR TITLE
Translate text

### PR DESCRIPTION
**Pull request by juliushaertl**
### 📝 Summary

- feat: Provide initial state for translations
- feat: Add translation modal to text

Can be tested with https://github.com/juliushaertl/integration_deepl

#### 🖼️ Screenshots

https://user-images.githubusercontent.com/3404133/222406598-e6ca443d-6eec-4441-a4b4-7cca3d1ca0bb.mov

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

**Comment by juliushaertl** [comment]
Needs a compile before the merge, ci is wrong here

File: src/components/Modal/Translate.vue [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+			<textarea v-model="input" />
+			<div class="language-selector">
+				<label for="fromLanguage">{{ t('text', 'Source language') }}</label>
+				<NcSelect v-model="fromLanguage"
+					input-id="fromLanguage"
+					:options="fromLanguages"
+					:append-to-body="false" />
+			</div>
+			<div class="language-selector">
+				<label for="fromLanguage">{{ t('text', 'Target language') }}</label>
```
Comment (marcelklehr) [inline src/components/Modal/Translate.vue:35]:
```suggestion
				<label for="toLanguage">{{ t('text', 'Target language') }}</label>
```

Comment (juliushaertl) [reply src/components/Modal/Translate.vue:35]: Good catch

File: src/components/Modal/Translate.vue [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+			error: null,
+		}
+	},
+	computed: {
+		fromLanguages() {
+			const result = this.canDetect ? [detectLanguageEntry] : []
+			const map = new Map()
+			for (const item of this.languages) {
+				if (!map.has(item.from)) {
+					map.set(item.from, true)
```
Comment (marcelklehr) [inline src/components/Modal/Translate.vue:105]: Using a Map as a Set?

**Review by marcelklehr — APPROVED** [review]

**Pull request merged**